test/partial.test.ts

```diff
@@ -50,4 +55,14 @@ describe('partial', () => {
 
     expectRejectValues(partialRt, rejectedValues)
   })
+
+  it('should preserve nonStrict', () => {
```

Comment:
I've added this test here and in the pick and omit tests. Not sure how far we want to go with testing non-strictness from both sides: e.g. how `nonStrict` uses / interacts with other runtypes, and how other runtypes use / interact with `nonStrict`

Reply:
Too me it looks like the right amount of testing.

Here you're checking that `omit` and `partial` are properly checking the `isNonStrict` flag/symbol and passing it when creating their `internalRecord`.

As simple-runtypes implementation is "just" functions calling functions, testing more complicated interactions is not necessary (as say opposed to a complicated runtype compiler). We know that if one function call works, nesting that will also work - that is the beauty of side-effect free functional code :grin:.